### ISSUE TITLE
imgui branch not building, probably due to leg/peg version mismatch

### ISSUE THREAD

**Issue by rib** [rib/gputop#168]
Trying to build the `imgui` branch I currently get this build failure:
```
FAILED: gputop-client-c/tracepoint_format.leg.h
/usr/bin/leg -P -o gputop-client-c/tracepoint_format.leg.h ../gputop-client-c/tracepoint_format.leg
/usr/bin/leg: invalid option -- 'P'
for usage try: /usr/bin/leg -h
```
`leg -h` for me shows:
```
leg version 0.1.15
usage: leg [<option>...] [<file>...]
where <option> can be
  -h          print this help information
  -o <ofile>  write output to <ofile>
  -v          be verbose
  -V          print version number and exit
if no <file> is given, input is read from stdin
if no <ofile> is given, output is written to stdout
```
On Arch Linux 0.1.15, from 2013, is the latest version of peg that's packaged (and even that is only available as a non-standard AUR package)

Is it maybe possible to build peg as a meson subproject() since it seems a little non-standard. A quick look at the [peg source tarball](http://piumarta.com/software/peg/peg-0.1.18.tar.gz) makes it look like it'd be quite straight-forward to build.

**Comment by rib** [comment]
This quick meson.build worked for me:
```
project('peg', 'c', version: '0.1.18', license: 'MIT')

leg_exe = executable('leg', [ 'src/leg.c', 'src/tree.c', 'src/compile.c' ],
                     c_args: [ '-Wno-unused-function', '-Wno-format-extra-args'])
```
but haven't tried hooking in as a subproject yet

**Comment by rib** [comment]
updating `gputop-client-c/meson.build` to find `log` with:
```
leg_sp = subproject('peg')
leg = leg_sp.get_variable('leg_exe')
```
worked for me and seems to have built `tracepoint_format.leg.h` (though I saw a lot of `-Wunused-parameter` warnings for `'yy'`, `'yytext'` and `'yyleng'`)

I have a different build failure now:
```
FAILED: subprojects/wslay-d72ea2ea9e24032144ab85f840e3aaf8cd5d6e07/lib/includes/wslay/wslayver.h
/home/bob/local/android-arm-toolchain-23/bin/python /home/bob/src/gputop/gputop-git-dev0/subprojects/wslay-d72ea2ea9e24032144ab85f840e3aaf8cd5d6e07/lib/includes/wslay/gen_wslayver.py --variable PACKAGE_VERSION=1.0.0 --input ../subprojects/wslay-d72ea2ea9e24032144ab85f840e3aaf8cd5d6e07/lib/includes/wslay/wslayver.h.in --output subprojects/wslay-d72ea2ea9e24032144ab85f840e3aaf8cd5d6e07/lib/includes/wslay/wslayver.h
Traceback (most recent call last):
  File "/home/bob/src/gputop/gputop-git-dev0/subprojects/wslay-d72ea2ea9e24032144ab85f840e3aaf8cd5d6e07/lib/includes/wslay/gen_wslayver.py", line 6, in <module>
    from mako.template import Template
ImportError: No module named mako.template
```
but guess I just need to pip install mako or something for that one

**Comment by djdeath** [comment]
Thanks, I'll tweak your PR for cross compiling.

**Comment by djdeath** [comment]
Fixed in 505269112cd3acca3884c62dc8d109499cec7e32.
